src/utils.rs

```diff
@@ -606,47 +666,169 @@ pub fn check_if_zero<'ctx>(
     Ok(flag.into())
 }
 
-/// Wrapper for calling the [`extend_memory`](crate::syscall::SyscallContext::extend_memory) syscall.
-pub(crate) fn extend_memory<'c>(
+pub(crate) fn compute_memory_cost<'c>(
     op_ctx: &'c OperationCtx,
     block: &'c Block,
-    new_size: Value<'c, 'c>,
+    memory_byte_size: Value<'c, 'c>,
 ) -> Result<Value<'c, 'c>, CodegenError> {
+    // this function computes memory cost, which is given by the following equations
+    // memory_size_word = (memory_byte_size + 31) / 32
+    // memory_cost = (memory_size_word ** 2) / 512 + (3 * memory_size_word)
+    //
+    //
     let context = op_ctx.mlir_context;
     let location = Location::unknown(context);
+    let uint32 = IntegerType::new(context, 32).into();
 
-    let ptr_type = pointer(context, 0);
+    let thirty_one = block
```

Comment:
Let's just use the number literal in the name. Same for the rest.
```suggestion
    let constant_31 = block
```